File: 1000/LC 357.go
func countNumbersWithUniqueDigits(n int) int {
    //0: 1
    //1: 9
    //2: 9 * 9
    if n == 0 {
        return 1
    }
    ans := 10
    cur := 9
    for i := 2; i <= n; i++ {
        cur *= 10 - i + 1
        ans += cur
    }

    return ans
}

File: 1000/LC 357.rs
impl Solution {
    pub fn count_numbers_with_unique_digits(n: i32) -> i32 {
        if n == 0 {
            return 1
        }

        let mut ans = 10;
        let mut cur = 9;
        for i in 2..=n {
            cur *= 10 - i + 1;
            ans += cur;
        }

        ans
    }
}